tool/state-processor/src/system/mod.rs

```diff
@@ -111,25 +123,101 @@ impl Processor {
 			};
 
 			if let Some(k) = try_get_evm_address(&key) {
+				if v.kton != 0 || v.kton_reserved != 0 {
+					let aa = AssetAccount {
+						balance: v.kton,
+						is_frozen: false,
+						reason: ExistenceReason::Sufficient,
+						extra: (),
+					};
+
+					a.sufficients += 1;
+					kton_details.accounts += 1;
+					kton_details.sufficients += 1;
+					// Note: this is double map structure in the pallet-assets.
+					self.shell_state.insert_value(
+						b"Assets",
+						b"Account",
+						&format!(
+							"{}{}",
+							array_bytes::bytes2hex("", blake2_128_concat(&KTON_ID.encode())),
+							array_bytes::bytes2hex("", blake2_128_concat(&k.encode())),
+						),
+						&aa,
+					);
+
+					// https://github.dev/darwinia-network/darwinia-common/blob/6a9392cfb9fe2c99b1c2b47d0c36125d61991bb7/frame/dvm/evm/precompiles/kton/src/lib.rs#L72
+					let mut approves = Map::<U256>::default();
+					self.solo_state.take_map(
+						b"KtonERC20",
+						b"Approves",
+						&mut approves,
+						get_hashed_key,
+					);
+					approves.iter().for_each(|(k, v)| {
+						kton_details.approvals += 1;
+						self.shell_state.insert_value(
+							b"Assets",
+							b"Approvals",
+							&k,
+							Approval { amount: v.as_u128(), deposit: 0 },
+						);
+					});
+				}
+
 				self.shell_state.insert_value(
 					b"System",
 					b"Account",
-					&array_bytes::bytes2hex("", subhasher::blake2_128_concat(k)),
+					&array_bytes::bytes2hex("", blake2_128_concat(k)),
```

Comment:
I hope to keep the function package prefix.
Easier for IDE to deduce the code while typing `subhasher::`.

Reply:
Got it. I remove this because that sometimes this affects the code format, for example:

```rs
array_bytes::bytes2hex(
	"",
	subhasher::blake2_128_concat(&KTON_ID.encode())
),
array_bytes::bytes2hex("", subhasher::blake2_128_concat(&k.encode())),
```

Please let me know if you think this is acceptable, I'll revert the changes.

Reply:
Yep, this is okay.